// azure-translator-code/dist/src/index.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.translateText = exports.translate = exports.updateTranslationsUnic = exports.updateTranslationsMulti = exports.translateToUnicFolder = exports.translateToMultipleFolders = void 0;
const translate_1 = __importDefault(require("./translate"));
exports.translate = translate_1.default;
const translateText_1 = __importDefault(require("./translate/translateText"));
exports.translateText = translateText_1.default;
const translateToMultipleFolders_1 = __importDefault(require("./translateToMultipleFolders"));
exports.translateToMultipleFolders = translateToMultipleFolders_1.default;
const translateToUnicFolder_1 = __importDefault(require("./translateToUnicFolder"));
exports.translateToUnicFolder = translateToUnicFolder_1.default;
const updateTranslationMulti_1 = __importDefault(require("./updateTranslationMulti"));
exports.updateTranslationsMulti = updateTranslationMulti_1.default;
const updateTranslationUnic_1 = __importDefault(require("./updateTranslationUnic"));
exports.updateTranslationsUnic = updateTranslationUnic_1.default;
